src/main/java/com/hubspot/jackson/datatype/protobuf/ProtobufEnumAsNumber.java

```diff
@@ -0,0 +1,79 @@
+package com.hubspot.jackson.datatype.protobuf;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.BeanProperty;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.deser.ContextualDeserializer;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ProtocolMessageEnum;
+
+public final class ProtobufEnumAsNumber {
+
+  private ProtobufEnumAsNumber() {}
+
+  public static class Serializer extends StdSerializer<ProtocolMessageEnum> {
+
+    protected Serializer() {
+      super(ProtocolMessageEnum.class);
+    }
+
+    @Override
+    public void serialize(ProtocolMessageEnum protocolMessageEnum, JsonGenerator jsonGenerator, SerializerProvider provider) throws IOException {
+      jsonGenerator.writeNumber(protocolMessageEnum.getNumber());
+    }
+  }
+
+  public static class Deserializer extends StdDeserializer<ProtocolMessageEnum> implements ContextualDeserializer {
+
+    private final JavaType javaType;
+
+    protected Deserializer() {
+      this(null);
+    }
+
+    protected Deserializer(JavaType javaType) {
+      super(ProtocolMessageEnum.class);
+      this.javaType = javaType;
+    }
+
+    @Override
+    public JsonDeserializer<?> createContextual(DeserializationContext context, BeanProperty beanProperty) {
+      // beanProperty is null when the type to deserialize is the top-level type or a generic type, not a type of a bean property
+      JavaType javaType = context.getContextualType();
+
+      if (javaType == null) {
+        javaType = beanProperty.getMember().getType();
+      }
+
+      return new Deserializer(javaType);
+    }
+
+    @Override
+    public ProtocolMessageEnum deserialize(JsonParser jsonParser, DeserializationContext context) throws IOException {
+      int intValue = jsonParser.getIntValue();
+      Class<?> enumClass = Preconditions.checkNotNull(javaType, "javaType").getRawClass();
+      try {
+        return (ProtocolMessageEnum) enumClass.getMethod("forNumber", int.class).invoke(null, intValue);
```

Comment:
Not sure if it's worth memoizing the class &rarr; method mapping, but I'm not super concerned about the performance implications of this implementation in the context I'd like to use this (deserializing from SQL via Rosetta).

Reply:
This could be moved into the `Deserializer` constructor but I don't remember off the top of my head whether contextual deserializers get cached automatically

Reply:
I moved it into `createContextual()`. 9d513ec